File: FrontEnd/Frontend_C#/SoftPac/SoftPacModel/DetallesFacturaDTO.cs
using SoftPac.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPac.Model
{
    [Serializable]
    public class DetallesFacturaDTO : EliminableDTOBase
    {
        private int? detalleFacturaId;
        private decimal subtotal;
        private String descripcion;
        private FacturasDTO factura;

        public int? DetalleFacturaId { get => detalleFacturaId; set => detalleFacturaId = value; }
        public decimal Subtotal { get => subtotal; set => subtotal = value; }
        public string Descripcion { get => descripcion; set => descripcion = value; }
        public FacturasDTO Factura { get => factura; set => factura = value; }

        public DetallesFacturaDTO() : base()
        {
            DetalleFacturaId = null;
            Subtotal = 0.0m;
            Descripcion = null;
            Factura = new FacturasDTO();
        }


        public DetallesFacturaDTO(int detalle_factura_id, decimal subtotal, string descripcion, FacturasDTO factura) : base()
        {
            this.DetalleFacturaId = detalle_factura_id;
            this.Subtotal = subtotal;
            this.Descripcion = descripcion;
            this.Factura = factura;
        }

        public DetallesFacturaDTO(DetallesFacturaDTO other) : base(other)
        {
            this.DetalleFacturaId = other.DetalleFacturaId;
            this.Subtotal = other.Subtotal;
            this.Descripcion = other.Descripcion;
            this.Factura = new FacturasDTO(other.Factura);
        }
    }
}

File: Backend_Java/SoftPacModel/src/main/java/pe/edu/pucp/softpac/model/DetallesFacturaDTO.java
package pe.edu.pucp.softpac.model;

import java.math.BigDecimal;

public class DetallesFacturaDTO extends EliminableDTOBase{
    private Integer detalle_factura_id;
    private BigDecimal subtotal;
    private String descripcion;
    private FacturasDTO factura;

    public DetallesFacturaDTO() {
        super();
        detalle_factura_id=null;
        subtotal=new BigDecimal("0.0");
        descripcion=null;
        factura=null;
    }

    public DetallesFacturaDTO(Integer detalle_factura_id, BigDecimal subtotal, String descripcion, FacturasDTO factura) {
        this.detalle_factura_id = detalle_factura_id;
        this.subtotal = subtotal;
        this.descripcion = descripcion;
        this.factura = factura;
    }
    
    public DetallesFacturaDTO(DetallesFacturaDTO detalle_factura){
        super(detalle_factura);
        this.detalle_factura_id = detalle_factura.getDetalle_factura_id();
        this.subtotal = detalle_factura.getSubtotal();
        this.descripcion = detalle_factura.getDescripcion();
        this.factura = new FacturasDTO(detalle_factura.getFactura());
    }

    /**
     * @return the detalle_factura_id
     */
    public Integer getDetalle_factura_id() {
        return detalle_factura_id;
    }

    /**
     * @param detalle_factura_id the detalle_factura_id to set
     */
    public void setDetalle_factura_id(Integer detalle_factura_id) {
        this.detalle_factura_id = detalle_factura_id;
    }
    /**
     * @return the descripcion
     */
    public String getDescripcion() {
        return descripcion;
    }

    /**
     * @param descripcion the descripcion to set
     */
    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
    /**
     * @return the subtotal
     */
    public BigDecimal getSubtotal() {
        return subtotal;
    }

    /**
     * @param subtotal the subtotal to set
     */
    public void setSubtotal(BigDecimal subtotal) {
        this.subtotal = subtotal;
    }

    public FacturasDTO getFactura() {
        return factura;
    }

    public  void setFactura(FacturasDTO factura) {
        this.factura = factura;
    }
}
